iothub/device/src/IotHubDeviceClient.cs

```diff
@@ -57,131 +59,44 @@ public IotHubDeviceClient(string hostName, IAuthenticationMethod authenticationM
         }
 
         private IotHubDeviceClient(IotHubConnectionCredentials iotHubConnectionCredentials, IotHubClientOptions options)
+            : base(iotHubConnectionCredentials, options)
         {
+            // Validate
             if (iotHubConnectionCredentials.ModuleId != null)
             {
                 throw new ArgumentException("A module Id was specified in the connection string - please use IotHubModuleClient for modules.");
```

Comment:
ArgumentException should specify the argument, however this isn't a parameter sent by the user. Maybe we should use InvalidOperationException instead?

Reply:
Yeah, I like that

Reply:
IotHubModuleClient, however, throws an ArgumentException/ArgumentNullException if the module Id isn't specified. For the sake of keeping exceptions similar, what do think about us throwing ArgumentException here and passing in the moduleId as arg name. Even though not set directly by the user, they do set it indirectly. The connection string flow might still be a bit confusing to corelate with this error though.

Reply:
Yeah, I'd rather not specify an argument that isn't immediately obvious to the user in context of the parameter names they passed.

Reply:
IOW, I think we should also change the module client to be consistent with this approach.